src/incubator/toast/helpers/useToastAnimation.ts

```diff
@@ -0,0 +1,64 @@
+import {useRef, useState} from 'react';
+import {Animated, Easing} from 'react-native';
+import {HapticService, HapticType} from 'services';
+import {ToastPreset, ToastProps} from '../types';
+
+interface UseToastAnimationProps extends Pick<ToastProps, 'visible' | 'preset' | 'position' | 'onAnimationEnd'> {
+  toastHeight: number;
+  playAccessibilityFeatures: () => void;
+  setDismissTimer: () => void;
+}
+
+export default ({
+  visible,
+  preset,
+  position,
+  toastHeight,
+  onAnimationEnd,
+  setDismissTimer,
+  playAccessibilityFeatures
+}: UseToastAnimationProps) => {
+  const toastAnimatedValue = useRef<Animated.Value>(new Animated.Value(0));
+  const [isAnimating, setIsAnimating] = useState<boolean>();
+
+  const _onAnimationEnd = () => {
+    if (visible) {
+      setDismissTimer();
+    } else {
+      setIsAnimating(false);
+    }
+
+    playAccessibilityFeatures();
+    onAnimationEnd?.(visible);
+  };
+
+  const toggleToast = (show = false, {delay}: {delay?: number} = {}) => {
+    Animated.timing(toastAnimatedValue.current, {
+      toValue: Number(show),
+      duration: 300,
+      delay,
+      easing: Easing.bezier(0.215, 0.61, 0.355, 1),
+      useNativeDriver: true
+    }).start(_onAnimationEnd);
+
+    if (preset === ToastPreset.FAILURE && show) {
+      HapticService.triggerHaptic(HapticType.impactMedium, 'Toast');
```

Comment:
I think this is a problem, the Haptic is an optional dependency, and the toast should be more generic.
We should either test here or create a function (or add an option to the current `triggerHaptic`) to only trigger if it's available and not trigger an error.

Reply:
Yes, you are right. 
I added a dedicate prop for that (`enableHapticFeedback`) and use that instead. 
In private we should enable it based on our logic